### ISSUE TITLE
No Tokens Even After Encrypted Backup & Restore

### ISSUE THREAD

**Issue by lalamover** [freeotp/freeotp-ios#234]
- FreeOTP is updated to the latest version (2.3) in my iPhone 5s iOS 12.4.3 (Model A1530)
- backed up using Macbook Pro 13-inch Mid 2012, Catalina 10.15.7 (MacBookPro9,2, i7, 2.9 GHz, 16GB RAM) with "Encrypt local backup" checked 
- backed up via iCloud (via Macbook Pro) with "Encrypt local backup" checked 
- backed up via iCloud (via iPhone 5s)
- restored to iPad Air iOS 12.5.3 (Model A1475) via Macbook Pro
- restored to iPad Air via iCloud
- restored to iPad Air via "iTunes"

The FreeOTP app is always restored to the iPad Air but it doesn't have any tokens inside T_T

Please feel free to let me know if I missed any detail or if anything is unclear.

**Comment by justin-stephenson** [comment]
Hi, when you open the App following restore does it just show an empty token list? Are you able to try restoring back to the phone, or a different iPhone?

**Comment by lalamover** [comment]
Thank you for replying Justin, I really appreciate it.

When I open the app, it just shows an empty token list, like the app's brand new.

I am about to try restoring it to an iPhone 6s. I will post an update soon.

Do you have any advice or suggestions on what I should do for now?

**Comment by justin-stephenson** [comment]
I need to test this on my device, encrypted backups + restores were working fine but I have not tested recently.

You may try adding some new tokens, take a new backup and see if any tokens get restored.

**Comment by lalamover** [comment]
I added a new token and took a new backup with my Macbook Pro.

I was going to restore it to iPhone 6s but I did a software update first to iOS 14.5.1.

Connected the iPhone 6s to my mac but it wouldn't read it even after restarting the iPhone and mac.

So I reset the iPhone (erase all content and settings).

Then at the “Apps & Data” Screen, I chose "Restore from iTunes Backup" and connected the iPhone 6s to my mac.

Voila! All my tokens now appeared!

So I tried the same process with my iPad Air. Couldn't upgrade its software, it's stuck at iOS 12.5.3.

After restore, I was still greeted with an empty tokens list.

So I guess it's either an iPad Air issue or an iOS 12.5.3 issue.

But I guess I'm good now as everything went fine with the iPhone 6s.

Thank you so much, Justin :)
